### PR TITLE
Introduce from_url to File module

### PR DESCRIPTION
**Pull request by kdbeall**
Thank you for the feedback in #13 !

### CODE REVIEW

File: opsmop/providers/file.py [hunk 1]
```diff
@@ -60,6 +60,16 @@ def should_replace_using_file(self):
 
     # ---------------------------------------------------------------
 
+    def should_replace_using_url(self):
+        """ for from_url, should we write the file? """
+        if not FileTests.exists(self.name):
+            return True
+        if not self.overwrite:
+            return False
+        return not FileTests.same_contents(self.name, self.from_url)
```
Comment (mpdehaan) [inline opsmop/providers/file.py]: In all this patch looks pretty good.  I think I see a small error here.  In this case, self.from_url will be a string path, so this test isn't going to match ever.

I would be ok with this for a first cut:

"return self.overwrite"

In the past when I was working on ansible we added an option where the resource could have a checksum and if the file didn't match the checksum it wouldn't download, which is useful for larger files, but if anybody wants that that can easily be a second PR.  It's also something most folks don't need.

File: opsmop/types/file.py [hunk 2]
```diff
@@ -38,8 +39,9 @@ def fields(self):
 
     def validate(self):
         v = Validators(self)
-        v.mutually_exclusive(['from_file', 'from_template', 'from_content'])
+        v.mutually_exclusive(['from_file', 'from_url', 'from_template', 'from_content'])
         v.mutually_exclusive(['directory', 'from_file'])
+        v.mutually_exclusive(['directory', 'from_url'])
```
Comment (mpdehaan) [inline opsmop/types/file.py]: this isn't part of your PR, but I just noticed it was still here, we don't need to do any tests against directory anymore as it is no longer an option to the module.  I can take care of this, but if you have a branch open if you want to delete lines containing 'directory' as part of this PR that would be super welcome :)

**Comment by kdbeall** [comment]
@mpdehaan I fixed the requested changes!

**Comment by mpdehaan** [comment]
Excellent, thank you!

I have some stuff going on in the same file module now to support push mode, so I'll get to merging this shortly!

**Comment by mpdehaan** [comment]
Sorry for the delay on this, I just completed basics on push mode so I can look at this when I get a second, but because of holiday stuff (and also grading papers, ha) this may be a little slow.  I don't anticipate needing any changes I just want to test it out and also add some docs to opsmop-docs before I push it.

**Comment by kdbeall** [comment]
@mpdehaan No worries. It took me longer than I expected as well! I was in Austin, TX for an interview.

**Comment by mpdehaan** [comment]
Got this merged in now, I made a small tweak in d0663ff where I was noticing it was *not* working with the integer timeout in the command array. I'm unclear how this worked on your python but I have confirmed from_url works fine now.  I am also pushing an example to the module docs shortly.

**Comment by mpdehaan** [comment]
Thanks!

**Pull request merged**